marklogic-data-hub/src/main/java/com/marklogic/hub/impl/DataHubImpl.java

```diff
@@ -405,6 +405,24 @@ public void clearUserModules(List<String> resourceNamesToNotDelete) {
         logger.info("Finished clearing user modules");
     }
 
+    public void clearUserArtifacts(){
+        final HubClient hubClientToUse = hubClient != null ? hubClient : hubConfig.newHubClient();
+
+        long start = System.currentTimeMillis();
+        logger.info("Clearing user artifacts as user: " + hubClientToUse.getUsername());
+
+        String query = "declareUpdate();\n" +
```

Comment:
We want as much logic in ML as possible via data services, so I think an ArtifactService.clearUserArtifacts method is the right approach here. 

Pinging @fsnow and @ryanjdew  for thoughts - I was thinking perhaps we should start a "datahub" service directory, where we have "global" methods like this. We also have a "system" service which just has getVersion in it. That arguably should be part of a datahub service. 

But we can punt on reorganizing things right now, as I think ArtifactService.clearUserArtifacts is reasonable. 

I think the DS endpoint should only clear that database that it's associated with - thus, this method should still exist, and it'll call clearUserArtifacts using both a staging client and a final client.

Reply:
I agree that is would be nice to compile a list needs like this from the Java middle-tier and move as much of it as possible to be server-side.